Hotfix v5.2.1 Release notes (#2534)
dotnet/SqlClient: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Stable release 5.2.1] - 2024-05-28

This update brings the below changes over the previous release:

### Changed

> - Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2528](https://github.com/dotnet/SqlClient/pull/2528)

Reviewer: Have you missed #2492? It can be included in this line as well.

> - Upgraded `Azure.Identity` version from 1.10.3 to 1.11.3 [#2492](https://github.com/dotnet/SqlClient/pull/2492), [#2528](https://github.com/dotnet/SqlClient/pull/2528)

Reviewer: This has more changes over upgrading Azure.Identity that I'd mention them here as well!

> - Code Health improvements: https://github.com/dotnet/SqlClient/pull/2467

Reviewer 2: follow markup language for all links. you can see samples at [here](https://github.com/dotnet/SqlClient/pull/2370/files).

> - Code Health improvements: [#2467](https://github.com/dotnet/SqlClient/pull/2467)

Reviewer 2: I think this needs to be moved under Fixed section. This is my opinion. What do you think @DavoudEshtehari

Reviewer: This fix hasn't changed the functionality, and I believe it's more appropriate under the current title.

Reviewer: This has more changes over upgrading Azure.Identity that I'd mention them here as well!

### Fixed

> - Fixed connection error from regression in SNIProxy code changes https://github.com/dotnet/SqlClient/pull/2436

Reviewer 3: ```suggestion
- Fixed connection errors on Linux when Data Source property contains both named instance and port https://github.com/dotnet/SqlClient/pull/2436
```

- Fixed `SqlConnection.FireInfoMessageEventOnUserErrors` when set to true throws an exception https://github.com/dotnet/SqlClient/pull/2505

> - Fixed DateTimeOffset size in TdsValueSetter.cs class file https://github.com/dotnet/SqlClient/pull/2506

Reviewer 3: ```suggestion
- Fixed exception when using DATETIMEOFFSET(n) in a TVP if n is 1, 2, 3, or 4 https://github.com/dotnet/SqlClient/pull/2506
```

> - Reverted the fix transient fault handling issue with OpenAsync https://github.com/dotnet/SqlClient/pull/2507

Reviewer 3: ```suggestion
- Reverted PR #1983 which caused connection failure delays when using `OpenAsync` https://github.com/dotnet/SqlClient/pull/2507
```

> - Fixed Clone of SqlConnection to include AccessTokenCallback https://github.com/dotnet/SqlClient/pull/2527

Reviewer 3: ```suggestion
- Fixed `SqlConnection.Clone()` to include `AccessTokenCallback` https://github.com/dotnet/SqlClient/pull/2527
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

> ## [Stable release 5.2.1] - 2024-05-28

Reviewer: ```suggestion
## [Stable release 5.2.1] - 2024-05-31
```
